src/core/sr.ml

```diff
@@ -141,9 +144,10 @@ let check_rule : sym StrMap.t -> sym * pp_hint * rule Pos.loc -> unit =
       fatal r.pos "Rule [%a] does not preserve typing." pp_rule (s,h,r.elt)
   | Some(cs) ->
   let is_constr c =
-    let eq_comm (t1,u1) (t2,u2) =
-      (Eval.eq_modulo t1 t2 && Eval.eq_modulo u1 u2) ||
-      (Eval.eq_modulo t1 u2 && Eval.eq_modulo t2 u1)
+    let eq_comm (c1,t1,u1) (c2,t2,u2) =
+      let ctx = match Ctxt.merge c1 c2 with None -> [] | Some(x) -> x in
```

Comment:
What's the interest of this merge function? It would perhaps be useful if there were defined variables in c1 and c2 but can `solve` generate contexts with defined variables? I don't think so. lhs_constrs are built from the empty context and there is no let in LHS.

Reply:
OK, I was not sure about it.